src/main/java/duke/sports/MyPlan.java

```diff
@@ -0,0 +1,374 @@
+package duke.sports;
+//import duke.data.Storage;
+import java.io.File;
+import java.io.FileNotFoundException;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.HashSet;
+import java.util.Set;
+
+ /**
+ * Loads a training plan from a txt file, create new plan, or edit a plan.
+ */
+
+public class MyPlan {
+
+     /**
+      * Represents the file path of the plans.
+      */
+    private String filePath;
+    /**
+     * Represents a scanner to take in the user input.
+     */
+    private Scanner fileInput;
+     /**
+      * Represents the list for the current loaded plan to be viewed or edited.
+      */
+    private ArrayList<MyTraining> list = new ArrayList<>();
+     /**
+      * Represents the list for the current number of plans saved.
+      */
+    private ArrayList<String> toc = new ArrayList<>();
+     /**
+      * Represents the map of all lists loaded from the text file.
+      */
+    private Map<String, ArrayList<MyTraining>> map = new HashMap<>();
+     /**
+      * Represents the name of the individual activity in a plan.
+      */
+    private String name;
+     /**
+      * Represents the number of sets of the activity in a plan.
+      */
+    private int sets;
+     /**
+      * Represents the number of repetitions of the activity in a plan.
+      */
+    private int reps;
+
+     /**
+      * The constructor for MyPlan.
+      * @throws FileNotFoundException if file is not found.
+      */
+    public MyPlan() throws FileNotFoundException {
+        filePath = ".\\src\\main\\java\\duke\\data\\plan.txt";
```

Comment:
Windows file path? Are you sure this works on all platforms? Refer to the [project constraints](https://nuscs2113-ay1920s1.github.io/website/admin/project-constraints.html), your product should be cross-platform compatible.